### ISSUE TITLE
Pre set mixes for self-stab receivers

### ISSUE THREAD

**Issue by Nicholas-Luoyi** [FrSkyRC/ETHOS-Feedback-Community#2792]
Here is another enhancement that we want to know if it is useful to add in ETHOS. We design to add new category while creating a model with wizard. If you select this category, you will only need to name the model and pick a image for you model. After the creation, ETHOS will automatically help you create all the necessary mixes for using a FrSky self-stab receiver. This will simplify the steps for new coming users. The attached image will show the pre set mixes.

![1690959329881_EE61239F-678F-4d0b-AFF4-A516B5ED929C](https://github.com/FrSkyRC/ETHOS-Feedback-Community/assets/58321103/ad887d1a-eb73-4525-ada2-eed18cdc95d5)

**Comment by mawzthefinn** [comment]
This would be awesome. 

Asked for in #1616 

A wizard option was requested in #1447

**Comment by dap35** [comment]
Great feature!  How will this support the differences between Archer/Archer plus stabilized RX and the Tandem stabilized RX?  I know the TD channels are different for gain and flight mode are different.  Also, TD by default is stabilizing 10 channels, so if you have flaps, it does some weird stuff until you disable the stabilization on the second group of 5 channels.

**Comment by mawzthefinn** [comment]
Archer Plus (except SR10+ on v1.0.2 only), Tandem, TW and RB25S/35S are all the same in terms of the implementation for Basic mode. TANDEM, TW and the two RB's also offer Advanced mode, Archer Plus does not (but Advanced is inherently not suitable for a wizard/pre-built mix as it will be user-defined)

ACCST, Archer and RB30/30+/40 are all the same in terms of implementation

**Comment by mawzthefinn** [comment]
Channels 7-11, not 6-10

Stabilizer section 1 is on ch 1,2,4,5 & 6
Stabilizer section 2 is on ch 7,8,9,10 & 11

In each case the first 3 channels in the set are fixed on or off by enabling or disabling the section.The last two (AIL2, ELE2 or AIL4, ELE4) are individually configurable. Except for the SR10+ on 1.0.2, I believe all the new receivers are defaulting to both sections on.

This is a receiver firmware default issue that you are commenting on, not related to the mixer setup at all.

**Comment by rburrow87** [comment]
Very nice addition. I have to check, but is it possible to re-run the wizard without it resetting everything? I ask because it's not uncommon to add a stabilizer later, so it would be nice to maybe have an SXR mix option in the add mix screen as well.

**Comment by BoltTed** [comment]
Does anyone have this issue with the SR10+ (I have latest update SW/Firmware/LUA also) where the stabilizer is acting in reverse for the elevator/ailerons. I did reverse it in the SRX STAB LUA setting for AIL Direction, or ELE Direction, but it is not working right. I just wondered if it was bc during SRX Calibration, the manual page or X20 screen is telling you to place pin right on receiver, but SR10+ is pin left, and things get reversed.

**Comment by mawzthefinn** [comment]
@BoltTed - with all the SR's, you have to set the reversing in the stabilizer first so that the stabilization is acting in the correct direction, then set direction in the radio so the radio mixer/stick direction is correct.

If you do it in the opposite order it will not work right.

**Comment by BoltTed** [comment]
I have a delta wing setup in the SRX Stable, but i could never get the ELEV to work correctly for stablization, when I moved the receiver (head of plane) up, both ailerons of the delta wing moved up. The stabilization for ailerons works, but not the elevator. I did a fresh model again with the model setup like first pic in this thread, calibrated the receiver SR10+ successfully, then setup the mixes for GAIN, MODES, EMERGENCY, then went to LUA for SRX Stable and turn ON stable mode, set to DELTA wing, then tried to get the ELEV to compensate when plane pitch up, but both ailerons moved up, whereas it should moved down. Even when I set the ELE Direction to REVERSE, it didn't work right either. Tried different combination of ELEV Direction and AIL Direction to NORMAL and REVERSE, and didn't work right either. Something is wrong with my SR10+ setup.

**Comment by Wolepo** [comment]
If you invert the effective direction of a channel, the corresponding outputs in the transmitter must also be inverted.

**Comment by Wolepo** [comment]
I could imagine that in the wizard another page is inserted, where the selection of: 
Aileron = 1 channel (CH1); 2 channels (CH1,CH5) or 4 channels (CH1, 5, 7, 8)
Elevator = 1 channel (CH2); 2 channels (CH2, CH6)
Rudder = 1 channel (CH4)
are offered.
In a 2nd page then the channels for sensitivity, gyro mode and self test are to be set.

**Comment by strgaltdel** [comment]
I would welcome more support for the Stabi systems "out of the box".

Personally, I would take a different approach:
I would add a mixer type to customize the typical Stabi settings and routes outputs to the corresponding channels provided by the respective Stabi system.

So for example a mixer "SRx" with mixer outputs to the above channels. 
(9,10,11, /12)

Within the mixer you can select sources for gain, mode switches, etc.

This is done for two reasons:

(1)
fewer mixers provide a better "overview" in the main view

(2)
better handling, the user does not have to surf back and forth between different mixers

(3)
in case you are in use of template supporting a classic Rx System, and you decide to update with a stabi system, you do only add this mixer, you dont't have to "start from zero" to customize the model by using a new category

(4)
Performance
my experience with huge templates is that Ethos scales rather poorly over the number of mixers / LSW's and SF's used
This is not (yet) a typical use case, but if more users develop complex templates with many stacked mixers imho performance bottlenecks will occure more and more.
This can possibly be countered by higher integrated mixers.

(i'm in preparation to open an issue with examples,
basically a "huge" template needs a lot of "priorized background tasks" which leads to bad performance handling in case you want to run several lua widgets.
This causes slow "cycling timig", in one case the log slowed down to 0.8 seconds.
audio output "chopped" a little bit
especially reading source values via lua (inputs / switches / telemetry) slowed down the behaviour.)

**Comment by bsongis-frsky** [comment]
Ready in 1.5.0 nightlies
